test/token/ERC721/extensions/ERC721Consecutive.test.js

```diff
@@ -0,0 +1,128 @@
+const { constants, expectEvent, expectRevert } = require('@openzeppelin/test-helpers');
+const { expect } = require('chai');
+
+const ERC721ConsecutiveMock = artifacts.require('ERC721ConsecutiveMock');
+
+contract('ERC721Consecutive', function (accounts) {
+  const [ user1, user2, user3, receiver ] = accounts;
+
+  const name = 'Non Fungible Token';
+  const symbol = 'NFT';
+  const batches = [
+    { receiver: user1, amount: 0 },
+    { receiver: user1, amount: 3 },
+    { receiver: user2, amount: 5 },
+    { receiver: user3, amount: 0 },
+    { receiver: user1, amount: 7 },
+  ];
+
+  beforeEach(async function () {
+    this.token = await ERC721ConsecutiveMock.new(
+      name,
+      symbol,
+      batches.map(({ receiver }) => receiver),
+      batches.map(({ amount }) => amount),
+    );
+  });
+
+  describe('minting during construction', function () {
+    it('events are emitted at construction', async function () {
+      let first = 0;
+
+      for (const batch of batches) {
+        if (batch.amount == 0) continue;
```

Comment:
We should check that the event was not emitted.

Reply:
I'd love to use `expectEvent.notEmitted.inConstruction`, but it doesn't try to match params. It just checks if an event with this name was emitted.

We would either have to improve `expectEvent.notEmitted.inConstruction` or do the decoding here. The `decodeLog` helper is not exported ...

Reply:
Is there no built in way to get the decoded events from the constructor?

Reply:
we have the txHash, so I guess we can get the tx receipt, and decode the events manually. It would bypass the helpers though